pkg/apis/core/validation/validation.go

```diff
@@ -4808,7 +4808,7 @@ func ValidateSecurityContext(sc *core.SecurityContext, fldPath *field.Path) fiel
 	}
 
 	if sc.RunAsUser != nil {
-		if *sc.RunAsUser < 0 {
+		if validation.IsValidUserID(*sc.RunAsUser) != nil {
 			allErrs = append(allErrs, field.Invalid(fldPath.Child("runAsUser"), *sc.RunAsUser, isNegativeErrorMsg))
```

Comment:
Could you use an error that is being returned from `IsValidUserID()` instead of `isNegativeErrorMsg` that doesn't cover all the error cases any more?

Reply:
done